### PR TITLE
docs: add documentation references

### PR DESCRIPTION
**Pull request by mattdibi**
This PR adds the "Documentation" section in the main repository README to help newcomers to the project find the information they need.

Additonally I removed the "Background" section since it's a bad fit for the main repository README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,19 +22,21 @@ Eclipse Kura™
...
 
 For more information, see the [Eclipse project proposal](http://www.eclipse.org/proposals/technology.kura/).
 
+Documentation
+-------------------
+
+- [**User Documentation**](https://eclipse.github.io/kura/latest/): here you'll find information on how to **use** Eclipse Kura i.e. installation instructions, informations on how to use the web UI and tutorials.
+- [**Developer Documentation**](https://github.com/eclipse/kura/wiki): the Eclipse Kura Github Wiki serves as a reference for **developers** who want to contribute to the Eclipse Kura project and/or develop new add-ons. Here you'll find Kura development/release model, guidelines on how to import internal packages, creating new bundles and development environment tips & tricks.
```
Comment (MMaiero) [inline README.md]: > Here you'll find Kura development/release model

Here you'll find the Eclipse Kura development/release model

File: README.md [hunk 2]
```diff
@@ -22,19 +22,21 @@ Eclipse Kura™
...
+- [**Docker Containers Documentation**](https://hub.docker.com/r/eclipse/kura/): the Eclipse Kura team also provides Docker containers for the project. Informations on how to build and run them are available at the project's Docker Hub page.
+- [**Developer Quickstart Guide**](https://github.com/eclipse/kura#getting-started): a quick guide on how to setup the development environment and build the project is also provided in this README.
+
+Additionally we provide two channels for reporting any issue you find with the project
```
Comment (MMaiero) [inline README.md]: > Additionally we provide two channels

Additionally, we provide two channels

File: README.md (hunk 1 above)
Comment (mattdibi) [reply README.md]: Fixed in f4dd22aed4f915e1df26c8e32a417a75ae996dd0

File: README.md (hunk 2 above)
Comment (mattdibi) [reply README.md]: Fixed in f4dd22aed4f915e1df26c8e32a417a75ae996dd0

**Review by MMaiero — APPROVED** [review]

**Comment by mattdibi** [comment]
Since this PR only affects the main README the CI failure is due to a flaky test. Merging anyway.

**Pull request merged**